webots/controllers/RL_Supervisor/networks.py

```diff
@@ -0,0 +1,126 @@
+"""Implementing Actor and Critic Networks"""
+
+# MIT License
+#
+# Copyright (c) 2023 - 2024 Andreas Merkle <web@blue-andi.de>
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+
+################################################################################
+# Imports
+################################################################################
+
+from tensorflow import keras  # pylint: disable=import-error
+from keras import layers  # pylint: disable=import-error
+from keras.regularizers import l2  # pylint: disable=import-error
+
+################################################################################
+# Variables
+################################################################################
+
+
+################################################################################
+# Classes
+################################################################################
+
+
+class Models:
```

Comment:
Consider deriving from tf.keras.Model to adhere to Keras architecture and match it's design principles.
https://www.tensorflow.org/api_docs/python/tf/keras/Model

Reply:
resolved with issue #156